tests/lib/types/dag-merge.nix

```diff
@@ -22,6 +22,9 @@ in {
         between =
           mkIf true (dag.entryBetween [ "after" ] [ "before" ] "between");
       }
+      { merged = "three"; }
+      { merged = dag.entryBetween [ "after" ] [ "before" ] "two"; }
+      { merged = dag.entryBefore [ "between" ] "one"; }
```

Comment:
I added more tests, including orderings on the `merged` values so that we don't rely on an unspecified ordering.

Note that `merged = mkAfter "foo";` would not work, first because it would hit https://github.com/NixOS/nixpkgs/pull/180701, but beyond that because it would order the *submodule* definition instead of the `data` definition. You'd need to do `merged = dag.entryAnywhere (mkAfter "foo");`.

Reply:
Okay, we can do `merged = mkAfter "foo";` now. Basically taking the useless `priority` from the submodule definition (`after` and `before` are not order-sensitive) and shoving it into `data`. This is probably deep into nobody-will-never-use-this territory, but it seems more correct and doesn't cost anything.